src/stash/src/postgres.rs

```diff
@@ -537,7 +581,18 @@ impl Postgres {
                                 }
                             }
                             attempt += 1;
-                            error!("tokio-postgres stash error, retry attempt {attempt}: {pgerr}");
+                            let cause = if let Some(code) = pgerr.code() {
```

Comment:
Is this the recommended way to use labels?

Reply:
Hm, so how many codes exist? The github xref thing here is a bit useless (points to the pgwire test runner). If this is less than a few dozen distinct values, it should be fine.

Reply:
On the backend, each unique label combination essentially acts as a metric. So e.g. 12 unique labels is exactly as expensive as adding 12 metrics. I'd consider a few dozen actually pretty high, personally, but that's backed by feels not numbers. If possible, I'd encourage you to see if you can somehow group things up into a smaller number of larger buckets (for example, expected vs suspicious). That said, if we think it's valuable to have specific numbers on each unique code, then a few dozen is certainly in the realm of "this will work".

Reply:
My gut is that the actual count will be 2 or less. If it ends up being more I'll bucket them somehow.

Reply:
👍